DaisyMicheal/component_library
# Feature/modal component

### Pull Request Description for Modal Component Library

**What does this PR do?**
Create a reusable Modal component as part of the React Component Library.

**Description of Task to be completed:**
- Create a Modal component with the following features:
  - A header section that includes a title and a close button.
  - A body section for the main content.
  - A footer section for actions like "Accept" or "Cancel".
  - Integration of a cookie SVG icon in the header.
- Ensure the modal has a responsive design and visually appealing layout.
- Implement a subtle background overlay to highlight the modal.
- Apply consistent styling for header, body, and footer sections.
- Control the modal visibility via the `showModal` state.
- Enable closing the modal by clicking the close button or outside the modal.
- Support props such as `show`, `onClose`, `title`, and `children` for customization.
- Write unit tests to cover the modal's functionality.
- Create Storybook stories to demonstrate the modal component.
- Update `App.js` to include an example of the modal in use.

**Changes Included:**
- Added `Modal.js` component file with the implementation.
- Styled the modal for consistent and responsive design.
- Integrated an SVG icon in the modal header.
- Implemented show/hide functionality for the modal.
- Created Storybook stories to demonstrate the modal component.
- Updated `App.js` to include an example of the modal in use.

**How should this be manually tested?**
1. Pull the latest changes from the branch.
2. Run the project using `npm start`.
3. Navigate to the Storybook interface to view and interact with the modal stories.
4. Verify the modal behavior in `App.js`.

**Any background context you want to provide?**
None

**What are the relevant pivotal tracker stories?**
[ #188013670](https://www.pivotaltracker.com/story/show/188013670)

**Screenshots (if appropriate):**
![image](https://github.com/user-attachments/assets/743bfd2b-42d9-4bda-bdf0-7e5a752930e2)
![image](https://github.com/user-attachments/assets/6a2ac95a-2db4-4d3a-82c5-a15eb1f1292b)

**Questions:**
None

**Additional Notes:**
- Ensure the modal is accessible with proper focus management and ARIA attributes.
- Test the modal with various content and states.
